mesatee_core/src/rpc/sgx/mod.rs

```diff
@@ -60,40 +60,47 @@ pub fn prelude() {
     ra::get_ra_cert();
 }
 
-pub type SgxMeasure = [u8; SGX_HASH_SIZE];
-
-// TODO: We hardcoded THREE auditors for now. Later we need to support arbitrary numbers of auditors.
-pub(crate) fn load_presigned_enclave_info() -> HashMap<String, (SgxMeasure, SgxMeasure)> {
+pub(crate) fn load_presigned_enclave_info() -> HashMap<String, EnclaveMeasurement> {
     use teaclave_config::ConfigSource;
+
+    #[cfg(not(feature = "mesalock_sgx"))]
+    use std::fs;
+    #[cfg(feature = "mesalock_sgx")]
+    use std::untrusted::fs;
+
     let ConfigSource::Path(ref enclave_info_path) = RUNTIME_CONFIG.audit.enclave_info;
-    let ConfigSource::Path(ref auditor_0_signature_path) =
-        RUNTIME_CONFIG.audit.auditor_signatures[0];
-    let ConfigSource::Path(ref auditor_1_signature_path) =
-        RUNTIME_CONFIG.audit.auditor_signatures[1];
-    let ConfigSource::Path(ref auditor_2_signature_path) =
-        RUNTIME_CONFIG.audit.auditor_signatures[2];
-
-    let enclave_signers: Vec<(&'static [u8], &std::path::Path)> = vec![
-        (
-            BUILD_CONFIG.auditor_public_keys[0],
-            &auditor_0_signature_path,
-        ),
-        (
-            BUILD_CONFIG.auditor_public_keys[1],
-            &auditor_1_signature_path,
-        ),
-        (
-            BUILD_CONFIG.auditor_public_keys[2],
-            &auditor_2_signature_path,
-        ),
-    ];
-
-    teaclave_utils::load_and_verify_enclave_info(&enclave_info_path, enclave_signers.as_slice())
+    let enclave_info_content = fs::read_to_string(enclave_info_path)
+        .unwrap_or_else(|_| panic!("Cannot find enclave info at {:?}.", enclave_info_path));
+
+    if RUNTIME_CONFIG.audit.auditor_signatures.len() < BUILD_CONFIG.auditor_public_keys.len() {
+        panic!("Number of auditor signatures is not enough for verification.")
+    }
+
+    let mut signatures: Vec<Vec<u8>> = vec![];
+    for ConfigSource::Path(ref path) in &RUNTIME_CONFIG.audit.auditor_signatures {
+        let signature =
+            fs::read(path).unwrap_or_else(|_| panic!("Cannot find signature file {:?}.", path));
+        signatures.push(signature);
+    }
+
+    for k in BUILD_CONFIG.auditor_public_keys {
```

Comment:
I am expecting the verify_enclave_info taking the for-loop logic into its scope.